sympy/concrete/summations.py

```diff
@@ -989,22 +991,21 @@ def eval_sum_symbolic(f, limits):
         r = gosper_sum(f, (i, a, b))
 
         # to find Sum at points where denominator approaches to zero
-        from sympy import together, fraction, Mul, Add, ordered
-        if isinstance(r, (Mul,Add)) and not(any(x in r.atoms(Symbol) for x in limits)):
-            var = ordered(r.atoms(Symbol))
-            _, denom = fraction(together(r))
-            if any(x in denom.atoms(Symbol) for x in var):
+        if isinstance(r, (Mul,Add)):
+            from sympy import ordered
+            var = r.free_symbols
+            var = [x for x in var if x not in limits]
+            den = denom(together(r))
+            if any(x in den.free_symbols for x in var):
                 cond = []
-                for v in var:
-                    for s in solve(denom, v):
+                for v in ordered(var):
+                    for s in solve(den, v):
                         m = Eq(v, s)
                         if not(m == False and m == True):
                             cond.append(m)
-                pic = []
-                for m in cond:
-                    pic.append((Sum(f_orig.subs(m.args[0], m.args[1]), limits).doit(),m))
-                pic.append((r, True))
-                r = Piecewise(*pic)
+                args = [(Sum(f_orig.subs(*m.args), limits).doit(), m) for m in cond]
```

Comment:
If there are 2 conditions (m1 and m2) it seems to me that you must add 3, not 2, cases: m1 & m2 True, m1 Trueand m2 True. So `args = [(Sum(f_orig.subs(dict(*[e.args for e in i])), limits).doit(), And(*i)) for i in reversed(list(subsets(cond))) if i]`

Reply:
(I had to do this sort of thing when integrating Piecewise functions with auxiliary conditions.)

Reply:
Case of more than one `cond` equal to true never came into my mind !
(btw that one line code is really cool :-) )